tony-azkaban/src/test/java/com/linkedin/tony/azkaban/TestTonyJob.java

```diff
@@ -63,6 +70,48 @@ public String getWorkingDirectory() {
     Assert.assertTrue(args.contains(TonyJobArg.SHELL_ENV.tonyParamName + " E1=e1"));
   }
 
+  /**
+   * Check if the flow level information is passed to the tony job through configuration.
+   */
+  @Test
+  public void testFlowInfoPropagation() {
+    final Props jobProps = new Props();
+    jobProps.put(TonyJobArg.HDFS_CLASSPATH.azPropName, "hdfs://nn:8020");
+    jobProps.put(CommonJobProperties.PROJECT_NAME, "unit_test");
+    jobProps.put(CommonJobProperties.FLOW_ID, "1");
+    jobProps.put(CommonJobProperties.EXEC_ID, "0");
+
+    final TonyJob tonyJob = new TonyJob("test_tony_job", new Props(), jobProps, log) {
+      @Override
+      public String getWorkingDirectory() {
+        return System.getProperty("java.io.tmpdir");
+      }
+    };
+    tonyJob.getMainArguments();
+
+    org.apache.hadoop.conf.Configuration conf =
+        new org.apache.hadoop.conf.Configuration(false);
+    conf.addResource(
+        new Path(tonyJob.getWorkingDirectory() + "_tony-conf-test_tony_job/tony.xml"));
+
+    String applicationTags = conf.get(TonyJob.TONY_APPLICATION_TAGS);
+    Assert.assertNotNull(applicationTags);
+
+    Set<String> values = new LinkedHashSet<>(
```

Comment:
Why a _Linked_HashSet as opposed to an ordinary `HashSet`? Also, instead of doing this parsing yourself, you can just use `conf.getStringCollection()`, which returns a `Collection<String>`.

Reply:
Addressed